vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -83,18 +140,34 @@ func NewInstallerController(
 func (c *InstallerController) createInstallerController(operatorSpec *operatorv1alpha1.OperatorSpec, originalOperatorStatus *operatorv1alpha1.StaticPodOperatorStatus, resourceVersion string) (bool, error) {
 	operatorStatus := originalOperatorStatus.DeepCopy()
 
+	if len(operatorStatus.NodeStatuses) == 0 {
+		return false, nil
+	}
+
+	// find upgrading node as this will be the first to start new deployment (to minimize number of down nodes)
+	startNode := 0
 	for i := range operatorStatus.NodeStatuses {
+		if operatorStatus.NodeStatuses[i].TargetDeploymentGeneration != 0 {
+			startNode = i
+			break
+		}
+	}
+
+	for l := 0; l < len(operatorStatus.NodeStatuses); l++ {
+		i := (startNode + l) % len(operatorStatus.NodeStatuses)
+		prev := (startNode + l - 1) % len(operatorStatus.NodeStatuses)
+
 		var currNodeState *operatorv1alpha1.NodeStatus
 		var prevNodeState *operatorv1alpha1.NodeStatus
 		currNodeState = &operatorStatus.NodeStatuses[i]
 		if i > 0 {
-			prevNodeState = &operatorStatus.NodeStatuses[i-1]
+			prevNodeState = &operatorStatus.NodeStatuses[prev]
 		}
 
 		// if we are in a transition, check to see if our installer pod completed
 		if currNodeState.TargetDeploymentGeneration > currNodeState.CurrentDeploymentGeneration {
-			// TODO check to see if our installer pod completed.  Success or failure there indicates whether we should be failed.
-			newCurrNodeState, err := c.newNodeStateForInstallInProgress(currNodeState)
+			pendingNewDeployment := operatorStatus.LatestAvailableDeploymentGeneration > currNodeState.TargetDeploymentGeneration
```

Comment:
This change bothers me.  Can you detect this a line earlier, avoid this branch, and just enter the "normal" new deployment flow?

Reply:
Tried that. We still want to wait for the installer to finish, otherwise they might be racing.